### PR TITLE
Add decision record for back-end tech stack 

### PR DESCRIPTION
**Pull request by alexsoble**
## What does this change? 

+ Adds ADR for backend tech stack

## What does this change look like?

[Preview link.](https://github.com/18F/OPRE-Unicorn/blob/ars/25/backend-tech-stack-adr/docs/adr/005-use-django-for-back-end-web-framework.md) 👁️ 

## GitHub cards

Completes part of #25.

### CODE REVIEW

**Comment by alexsoble** [comment]
@amymok @carjug Ready for review; comments // suggestions welcome!

**Review by carjug — APPROVED** [review]
Looks good to me! Your explanation of web frameworks is 💯 very good.

**Comment by alexsoble** [comment]
Awwwww thank you @carjug! 😊

**Review by amymok — APPROVED** [review]
This looks really awesome, thank you so much!

**Comment by alexsoble** [comment]
@ninamak During coworking today you had some verbal feedback on this ADR — would you mind leaving any notes on the PR here so that I can incorporate them, when you get a chance? 

Thanks!

File: docs/adr/005-use-django-for-back-end-web-framework.md
Comment (ninamak) [inline docs/adr/005-use-django-for-back-end-web-framework.md:27]: There was a lot of helpful content in the comments thread on Issue #25, some of which would be helpful to include here to provide more context about what these features are.  For example: 

> Features that can be used out of the box for this application without too much development work needed:
> 
> - ORM: since there is a need for a database, each Django project will come with the configuration to a database
> - Authentication/Authorization: There will be some form of authentication/authorization needed, Django already comes with this package, and more if we need to use more specific authentication
> - Admin interface: This will be good for administrator to edit data easily

Also the maturity of Django, TTS's familiarity/success with it, etc.

Comment (ninamak) [inline docs/adr/005-use-django-for-back-end-web-framework.md:27]: In the comments thread, Carly had left this comment, and wondering what came of it, esp given the uncertainty we still have around whether we will be going down the single page web app route.

> I've used Flask and have found it to be very easy to use with lots of reliable and readable documentation. It does take a bit more work than Django to get a Flask app connected to a database, but it's doable no doubt. It's very lightweight and would suggest this if we go the single page web app route.

File: docs/adr/005-use-django-for-back-end-web-framework.md
Comment (ninamak) [inline docs/adr/005-use-django-for-back-end-web-framework.md:25]: Can we add a section to explicitly call out options considered, and maybe the factors considered as well?

**Review by ninamak — COMMENTED** [review]
@alexsoble I left a few comments to capture the verbal feedback I offered yesterday.  Not looking to add a ton more effort to this doc, just a few mins to make it more complete

File: docs/adr/005-use-django-for-back-end-web-framework.md
Comment (alexsoble) [reply docs/adr/005-use-django-for-back-end-web-framework.md:27]: Gotcha! I included a link to the Issue in the ADR, so readers can always click through and read more. I can copy over some of those bullets into the ADR if that would be helpful. As a directional note, I like to think about ADRs as lightweight journal entires — see the [example ADR in the recent 18F blog post](https://18f.gsa.gov/2021/07/06/architecture_decision_records_helpful_now_invaluable_later/#whats-an-adr) for en example of the length & depth of a typical ADR.

**Comment by alexsoble** [comment]
@ninamak Took a pass at incorporating your feedback, [please see the latest](https://github.com/18F/OPRE-Unicorn/blob/ars/25/backend-tech-stack-adr/docs/adr/005-use-django-for-back-end-web-framework.md)! I think these additional sections make the end product stronger. I think we're moving somewhat away from lightweight ADRs towards more comprehensive decision memos, which is OK if that's where we want to go as a team.

**Comment by alexsoble** [comment]
@ninamak Quick reminder to re-review and approve this one when you have sec! I know it's end of day east coast, this can of course wait til tomorrow.

**Review by ninamak — APPROVED** [review]
Looks great to me!

**Comment by ninamak** [comment]
Let's keep striving for lightweight, but with an eye towards "what is most important to communicate to someone new to this project or someone reviewing decisions," assuming we will not be around forever to explain.  No one is going to read or value dense memos, and you're right to want to stay away from that.

**Comment by alexsoble** [comment]
@ninamak Wanna hit the merge button? 😄 

Or feel free to log off and let another teammate hit "merge", it's late East Coast!

(We have a "no merging your own PRs" preference!)

**Comment by alexsoble** [comment]
I know we don't generally merge our own PRs, but this has 3 👍 approvals, so I'm going to go for it! 🚢

**Pull request merged**
